register-machines/src/main.rs

```diff
@@ -1,8 +1,10 @@
-use std::collections::HashMap;
+use std::{collections::HashMap, vec};
+use rug::Integer;
 
+type Godel = Integer; 
```

Comment:
Godel numbers are defined only for Natural numbers. This obviously isn't a problem for this task, but something worth noting nonetheless.